src/exchange-connector/resourceLoader/fatRamlResourceLoader.ts

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2020, salesforce.com, inc.
+ * All rights reserved.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import amf from "amf-client-js";
+import path from "path";
+import { FileSystemAdapter } from "./fileSystemAdapter";
+
+const EXCHANGE_MODULES = "exchange_modules/";
+
+export class FatRamlResourceLoader implements amf.resource.ResourceLoader {
+  public workingDir: string;
+  public fsAdapter: FileSystemAdapter;
+
+  constructor(rootFolder: string) {
+    this.workingDir = rootFolder.split("file://")[1];
+    if (!this.workingDir) {
+      throw new Error("rootFolder does not contain 'file://' prefix");
+    }
+    this.fsAdapter = new FileSystemAdapter();
+  }
+
+  accepts(resource: string): boolean {
+    return resource?.indexOf(EXCHANGE_MODULES) >= 0;
+  }
+
+  fetch(resource: string): Promise<amf.client.remote.Content> {
+    return new Promise((resolve, reject) => {
+      if (resource.indexOf(EXCHANGE_MODULES) >= 0) {
```

Comment:
Not really resilient as this would find exchange_modules anywhere in the. path and move it to the top level.  Not sure if that is what you want or if you want just if there is exchange_modeles at the beginning of the resource path instead of anywhere.